### ISSUE TITLE
ml_clustering_evaluator() returns error if used with ml_cross_validator()

### ISSUE THREAD

**Issue by benmwhite** [rstudio/sparklyr#1936]
`ml_clustering_evaluator()` works fine on single prediction tables, but fails when used as the evaluator within `ml_cross_validator()` or `ml_train_validation_split()`. Basing the reprex on the working example from [this RStudio blog post from last year](https://blog.rstudio.com/2018/05/14/sparklyr-0-8/) which used `ml_multiclass_classification_evaluator()`.

Verifying single table case:
```
library(sparklyr)
sc <- spark_connect(master = "local", version = "2.4.0")
dplyr::copy_to(sc, iris, "iris_tbl")


pipeline <- ml_pipeline(sc) %>%
  ft_vector_assembler(
    input_cols = c("Sepal_Width", "Sepal_Length", "Petal_Width", "Petal_Length"),
    output_col = "features"
  ) %>%
  ml_kmeans(k = 3, uid = "kmeans")
```
The following works as expected:

```
ml_fit_and_transform(pipeline, dplyr::tbl(sc, "iris_tbl")) %>% 
  ml_clustering_evaluator()
```
This returns the average silhouette score for the single K-means model with k=3. 

However if we try to use `ml_cross_validator()` over the `k` parameter with a similar pipeline then it fails:
```
pipeline_cv <- ml_pipeline(sc) %>%
  ft_vector_assembler(
    input_cols = c("Sepal_Width", "Sepal_Length", "Petal_Width", "Petal_Length"),
    output_col = "features"
  ) %>%
  ml_kmeans(uid = "kmeans")

# Specify hyperparameter grid
grid <- list(
  kmeans = list(
    k = c(2, 3, 4)
  )
)

# Create the cross validator object
cv <- ml_cross_validator(
  sc, 
  estimator = pipeline_cv, 
  estimator_param_maps = grid,
  evaluator = ml_clustering_evaluator(sc),
  num_folds = 3, 
  parallelism = 1
)

# Train the models
cv_model <- ml_fit(cv, dplyr::tbl(sc, "iris_tbl"))
```

This returns the error `Error in spark_connection(jobj) :
  argument "jobj" is missing, with no default`. The same error occurs for `ml_train_validation_split()`.

Session info:

```
R version 3.4.1 (2017-06-30)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Amazon Linux AMI 2018.03

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] sparklyr_1.0.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0       rstudioapi_0.9.0 magrittr_1.5     tidyselect_0.2.5
 [5] R6_2.4.0         rlang_0.3.1      fansi_0.4.0      httr_1.4.0
 [9] dplyr_0.8.0.1    tools_3.4.1      parallel_3.4.1   config_0.3
[13] utf8_1.1.4       cli_1.0.1        DBI_1.0.0        withr_2.1.2
[17] dbplyr_1.3.0     askpass_1.1      htmltools_0.3.6  ellipsis_0.1.0
[21] openssl_1.2.1    yaml_2.2.0       assertthat_0.2.0 rprojroot_1.3-2
[25] digest_0.6.18    tibble_2.0.1     forge_0.2.0      crayon_1.3.4
[29] purrr_0.3.0      base64enc_0.1-3  htmlwidgets_1.3  glue_1.3.0
[33] compiler_3.4.1   pillar_1.3.1     generics_0.0.2   r2d3_0.2.3
[37] backports_1.1.3  jsonlite_1.6     pkgconfig_2.0.2

```

**Comment by kevinykuo** [comment]
Thanks for the report! Let me know if you run into further issues with this.

**Comment by sgemerek** [comment]
has this issue been resolved?  and, if so, in what version of sparklyr?

**Comment by benmwhite** [comment]
@sgemerek I haven't had any issues since this was patched in version 1.0.5, so if you're using the latest CRAN version (1.1.0) you should be fine. See the session info in the first post and the [pull request reference](https://github.com/sparklyr/sparklyr/commit/dca07229a64fbecf0e4c8adec45aae8b59ab6b08) when @kevinykuo patched it last March.
